File: src/main/java/net/flyingfishflash/loremlist/domain/lrmlist/LrmListRepository.java
package net.flyingfishflash.loremlist.domain.lrmlist;

import java.util.List;
import java.util.Set;
import java.util.UUID;

public interface LrmListRepository {
  long countByOwner(String owner);

  int delete();

  int deleteById(Set<UUID> ids);

  int deleteByOwnerAndId(UUID id, String owner);

  List<LrmList> findByOwner(String owner);

  LrmList findByOwnerAndIdOrNull(UUID id, String owner);

  List<LrmList> findByOwnerAndHavingNoItemAssociations(String owner);

  List<LrmList> findByPublic();

  List<UUID> findIdsByOwnerAndIds(List<UUID> listIdCollection, String owner);

  Set<UUID> notFoundByOwnerAndId(List<UUID> listIdCollection, String owner);

  UUID insert(LrmList lrmList);

  int update(LrmList lrmList);

  int updateName(LrmList lrmList);

  int updateDescription(LrmList lrmList);

  int updateIsPublic(LrmList lrmList);
}

File: src/main/kotlin/net/flyingfishflash/loremlist/domain/lrmlist/LrmListRepository.kt
package net.flyingfishflash.loremlist.domain.lrmlist

import java.util.UUID

interface LrmListRepository {
  fun countByOwner(owner: String): Long
  fun delete(): Int
  fun deleteById(ids: Set<UUID>): Int
  fun deleteByOwnerAndId(id: UUID, owner: String): Int
  fun findByOwner(owner: String): List<LrmList>
  fun findByOwnerAndIdOrNull(id: UUID, owner: String): LrmList?
  fun findByOwnerAndHavingNoItemAssociations(owner: String): List<LrmList>
  fun findByPublic(): List<LrmList>
  fun findIdsByOwnerAndIds(listIdCollection: List<UUID>, owner: String): List<UUID>
  fun notFoundByOwnerAndId(listIdCollection: List<UUID>, owner: String): Set<UUID>
  fun insert(lrmList: LrmList, subject: String): UUID
  fun update(lrmList: LrmList): Int
  fun updateName(lrmList: LrmList): Int
  fun updateDescription(lrmList: LrmList): Int
  fun updateIsPublic(lrmList: LrmList): Int
}
